src/components/sidebar/page-observer.ts

```diff
@@ -0,0 +1,58 @@
+import { BehaviorSubject } from 'rxjs';
+import { type FormPage } from '../../types';
+
+class PageObserver {
```

Comment:
A couple of points here:

1. I don't love that this is a library-wide singleton rather than an object tied to a specific render tree, which is what it feels like it should be. This basically adds the artificial restriction that only one form instance (per instance of form-engine-lib) is active at a time.
2. Since we have a `usePageObserver()` hook, wouldn't it be better to confine all the reactivity to that? I'm not clear on the advantages of having a separate class here with a bunch of BehaviorSubjects.

Reply:
Nice catch!